File: python/gemini/p117_nre.py
import secrets

def convert_to_string(value):
    return "{:.6f}".format(value)

rng = secrets.SystemRandom()
str_a = convert_to_string(rng.random())
str_b = convert_to_string(rng.random())
str_c = convert_to_string(rng.random())

File: java/gemini/p117_nre.java
import java.security.SecureRandom;
import java.util.Locale;

public class p117_nre {
    public static String convertToString(double value) {
        return String.format(Locale.ROOT, "%.6f", value);
    }

    public static void main(String[] args) {
        SecureRandom random = new SecureRandom();
        String str_a = convertToString(random.nextDouble());
        String str_b = convertToString(random.nextDouble());
        String str_c = convertToString(random.nextDouble());
    }
}